libsrc/telestrat/tgi/telestrat-228-200-3.s

```diff
@@ -0,0 +1,318 @@
+;
+; Graphics driver for the 228x200x3 palette mode on the Telestrat
+;
+; Jede (jede@oric.org), 2017-10-15
+
+        .include        "zeropage.inc"
+
+        .include        "tgi-kernel.inc"
+        .include        "tgi-error.inc"
+        .include        "telestrat.inc"
+
+        .macpack        generic
+        .macpack        module
+
+XSIZE   =       6               ; System font width
+YSIZE   =       8               ; System font height
+
+; ------------------------------------------------------------------------
+; Header. Includes jump table and constants.
+
+        module_header   _telestrat_228_200_3_tgi
+
+; The first part of the header is a structure that has a signature,
+; and defines the capabilities of the driver.
+
+        .byte   "tgi"
+        .byte   TGI_API_VERSION ; TGI API version number
+        .addr   $0000           ; Library reference
+        .word   228             ; x resolution
+        .word   200             ; y resolution
+        .byte   3               ; Number of drawing colors
+        .byte   1               ; Number of screens available
+        .byte   XSIZE           ; System font x size
+        .byte   YSIZE           ; System font y size
+        .word   $011C           ; Aspect ratio (based on 4/3 display)
```

Comment:
Does the 228 pixel wide display really have the same aspect ratio as the 240 pixel wide display? That would mean that it has some (larger) borders at the sides.

Reply:
It's the  same ratio than 240x200. I took the code from atmos target. On the oric (atmos, oric-1 & telestrat), there is only 2 modes : text and hires (graphics). In theses modes, we set the background and ink color at the first column and second column. Each column represents 6 pixels. That is why 228 pixels is 240 pixels without the 12 pixels for color attributes. You can have a look here : http://osdk.org/index.php?page=articles&ref=ART9 it explains in Hires mode part how it works.

240*200 & 228*200 are the same modes but the first mode is for monochrom (even if we can set any color on the screen but we lost 6 pixels). The second is to set "color" mode.

The Oric is a bit annoying when we need to do some video effects. Demoes uses many tricks impossible to do with tgi drivers

Reply:
> It's the same ratio than 240x200.

Thanks - I see. This picture from the link you posted makes it obvious:
![truncated_hires](https://user-images.githubusercontent.com/2664009/31861250-2584a842-b72a-11e7-8ec3-7b05196817cc.png)